### PR TITLE
MGDOBR-1180 - Always export junit artifacts

### PR DESCRIPTION
**Pull request by jiripetrlik**
Forgot to export junits also for failed runs.

### CODE REVIEW

File: .github/workflows/dev-cluster-check.yml [hunk 1]
```diff
@@ -83,6 +83,7 @@ jobs:
           working-directory: integration-tests
       - name: Archive junit test results
         uses: actions/upload-artifact@v3
+        if: ${{ always() }}
```
Comment (afalhambra) [inline .github/workflows/dev-cluster-check.yml:86]: I think we don't need curly braces here, wdyt?

Comment (sutaakar) [reply .github/workflows/dev-cluster-check.yml:86]: Seems, so, though it is used in examples
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif

**Review by sutaakar — APPROVED** [review]

File: .github/workflows/dev-cluster-check.yml (hunk 1 above)
Comment (afalhambra) [reply .github/workflows/dev-cluster-check.yml:86]: we have a mix in our workflows, some with curly [braces](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/.github/workflows/dev-cluster-check.yml#L109) and some without [them](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/.github/workflows/dev-cluster-check.yml#L117)

**Review by afalhambra — APPROVED** [review]

File: .github/workflows/dev-cluster-check.yml (hunk 1 above)
Comment (jiripetrlik) [reply .github/workflows/dev-cluster-check.yml:86]: Good point. I can look at it and try to unify this in the separate PR.

**Pull request merged**